iceoryx_posh/test/moduletests/test_roudi_iceoyx_roudi_memory_manager.cpp

```diff
@@ -76,5 +76,47 @@ TEST_F(IceoryxRoudiMemoryManager_test, createAndAnnouceMemorySuccess)
     EXPECT_THAT(testResult.has_error(), Eq(false));
 }
 
+/// @brief test to check function MgmtMemoryProviderSuccess
+TEST_F(IceoryxRoudiMemoryManager_test, MgmtMemoryProviderSuccess)
+{
+    auto testResult = m_roudiMemoryManagerTest->mgmtMemoryProvider();
+    EXPECT_THAT(testResult, Not(Eq(nullptr)));
+}
+
+/// @brief test to check function NullOptCheck
+TEST_F(IceoryxRoudiMemoryManager_test, NullOptCheck)
```

Comment:
You are testing here multiple things. We follow the arrange, act, assert pattern.
See this for an introduction: https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80

Could you please split up this test into several tests named something like:
`AcquiringIntrospectionMemoryManagerAfterCreateAndAnnounceMemoryIsSuccessful` I know it is a long name but everyone then know what the test does and what the expected outcome is.